.github/workflows/workflow.yml

```diff
@@ -2,149 +2,102 @@ name: workflow
 
 on: [push, pull_request]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   lock:
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
```

Comment:
I'm not too fond of this solution, but there is a way to set this once using repository variables. You set it [in the github UI](https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repository), similar to setting repository secrets. Then you can access it from anywhere in your workflow file like so:
```yml
python-version: ${{ vars.<YOUR_VAR_NAME> }}
```

Reply:
🤔 so that way we could add/remove versions without changing the workflow file, i can see how this could be useful. But I think I prefer the file to be as self-explanatory as possible

Reply:
Yes exactly. Pros and cons.